lib/src/main/java/co/nimblehq/alpine/lib/AlpineLib.kt

```diff
@@ -0,0 +1,55 @@
+package co.nimblehq.alpine.lib
+
+import android.content.Context
+import android.nfc.Tag
+import co.nimblehq.alpine.lib.model.CameraImage
+import co.nimblehq.alpine.lib.model.MrzInfo
+import co.nimblehq.alpine.lib.mrz.MrzProcessor
+import co.nimblehq.alpine.lib.mrz.MrzProcessorImpl
+import co.nimblehq.alpine.lib.mrz.MrzProcessorResultListener
+import co.nimblehq.alpine.lib.nfc.ISO_DEP_TIMEOUT_IN_MILLIS
+import co.nimblehq.alpine.lib.nfc.NfcReader
+import co.nimblehq.alpine.lib.nfc.NfcReaderImpl
+import com.google.mlkit.vision.common.InputImage
+
+object AlpineLib {
+
+    private val nfcReader: NfcReader = NfcReaderImpl()
+    private val mrzProcessor: MrzProcessor = MrzProcessorImpl()
+
+    fun initializeNfcReader(context: Context) = nfcReader.initialize(context)
```

Comment:
What if we remove this method and have the `readNfc()` methods take a `Context` instead? Having an `initialize()` method could be a pain point for end user if they forget to initialize. Having `Context` in the method being used is more explicit IMO. What do you think?

Reply:
@ryan-conway Actually, We should recommend user `initialize` in the `Application` class, but we don't have it for our `Sample.` 🙏  the context should be from the `Application` level, What do you think?

Reply:
We can handle that inside the library anyway by using `context.applicationContext` 😉

Reply:
Yeah @ryan-conway, we can use `context.applicationContext`  but What about if the user needs to call `readNfc` somewhere else that's not activity e.g.,  in `VM` or `UseCase` 🤔  As `VM` we should try to avoid passing the `context` inside there. So I think it's better to `initialize` when `application` is created. It seems like having a separate method for `initialize` is more flexible 🙇‍♀️ What do you think?

Reply:
If that's the case why don't we make `AlbineLib` into a regular class with `context` in the `constructor` instead? Then the user can just inject the class wherever they need it, and we don't have to deal with the [singleton problem](https://www.vojtechruzicka.com/singleton-pattern-pitfalls/#:~:text=One%20of%20the%20main%20disadvantages,test%20the%20Singleton%20as%20well.). Should be better in terms of garbage collection too since it doesn't necessarily need to survive for the whole lifecycle of the app, what do you think?

Reply:
> If that's the case why don't we make AlbineLib into a regular class with context in the constructor instead?

Can you show me some examples to me 🙏

Reply:
@Wadeewee How about something like this this? 👀 
```kotlin
class AlpineLib(context: Context) {
    private val nfcReader: NfcReader = NfcReaderImpl(context)
    private val mrzProcessor: MrzProcessor = MrzProcessorImpl()

    fun readNfc(tag: Tag) { /* */ }
    fun processImageFile(filePath: String) { /* */ }
}

Reply:
Oh, @ryan-conway, as I discussed with @AVI5HEK this morning, I will reverse the previous one. Users should call our library only when they need to `readNfc` of `proceesImage` 🙏 🚀